julia-package/brainflow/src/board_shim.jl

```diff
@@ -229,8 +229,13 @@ end
     return sub_string
 end
 
-@brainflow_rethrow function get_board_data(board_shim::BoardShim)
+@brainflow_rethrow function get_board_data(board_shim::BoardShim, num_samples::Int64=1)
```

Comment:
1 is bad default because it changes the behavior of existing code. Currently users have `get_board_data()` and it returns all available data, such behavior should not be changed. Set default to -1 for example or to 0 and handle it to return all data.

Or add new methods if its allowed in Julia(I don't remember, you can check or ask @matthijscox )